doc/source/cd/background.ipynb

```diff
@@ -0,0 +1,608 @@
+    "\n",
+    "1. Linear projections\n",
+    "2. Auto-encoders\n",
+    "3. Black-box shift detectors\n",
+    "\n",
+    "[Alibi Detect](https://github.com/SeldonIO/alibi-detect) allows for a high degree of flexibility here, with a user's chosen dimension reduction technique able to be incorperated into their chosen detector via the `preprocess_fn` and `preprocess_x_ref` arguments. In the following sections, the three categories of technqiues are briefily introduced. For more details, see the [examples](examples.rst). \n",
+    "\n",
+    "\n",
+    "#### Linear projections \n",
+    "\n",
+    "This includes dimension reduction techniques such as [principal component analysis (PCA)](https://en.wikipedia.org/wiki/Principal_component_analysis) and [sparse random projections (SRP)](https://doi.org/10.1145/1150402.1150436). These techniques involve using a transformation or projection matrix $\\mathbf{R}$ to reduce the dimensionality of a given data matrix $\\mathbf{X}$, such that $\\hat{\\mathbf{X}} = \\mathbf{XR}$. A straightforward way to include such techniques as a pre-processing stage is to pass them to the detectors via the `preprocess_fn` argument, for example for the `scikit-learn` library's `PCA` class: \n",
+    "\n",
+    "```python\n",
+    "pca = PCA(2)\n",
+    "pca.fit(X_train)\n",
+    "detector = MMDDrift(X_train, backend='tensorflow', p_val=.05, preprocess_fn=pca.transform)\n",
+    "```\n",
+    "\n",
+    "#### Auto-encoders \n",
+    "\n",
+    "These non-linear dimension reducing representations consist of an encoder function $\\phi : \\mathcal{X} \\mapsto \\mathcal{H}$ and a decoder function $\\psi : \\mathcal{H} \\mapsto \\mathcal{X}$, where the latent space $\\mathcal{H}$ has lower dimensionality than the input space $\\mathcal{X}$. The output of the encoder $\\hat{\\mathbf{X}} \\in \\mathcal{H}$ can then be monitored by the drift detector. Training involves learning both the encoding function $\\phi$ and the decoding function $\\psi$, in order to reduce the reconstruction loss: $\\phi, \\psi = \\text{arg} \\min_{\\phi, \\psi}\\, \\lVert \\mathbf{X}-(\\phi \\circ \\psi)\\mathbf{X}\\rVert^2$. However, either trained (TAE) or untrained autoencoders (UAE) can be used. A `tensorflow` autoencoder can be incorporated into a detector by packaging it as a callable function using `preprocess_drift` (from [alibi_detect.cd.tensorflow.preprocess](../api/alibi_detect.cd.tensorflow.preprocess.rst)) and `partial` (see [here](https://docs.python.org/3/library/functools.html#functools.partial)), for example:\n",
```

Comment:
Would highlight that MSE is just an example of a training loss used since there are multiple ways to train an AE. Also the similar comment for `X_train` as before, making sure that `X_ref` is used everywhere and should not overlap with `X_train` for pretrained preprocessing functions.